src/backend/mailbox_manager/models.py

```diff
@@ -303,7 +303,15 @@ class Meta:
         db_table = "people_mail_box"
         verbose_name = _("Mailbox")
         verbose_name_plural = _("Mailboxes")
-        unique_together = ("local_part", "domain")
+        constraints = [
+            models.UniqueConstraint(
+                fields=["local_part", "domain"], name="unique_username"
+            ),
+            models.UniqueConstraint(
+                fields=["first_name", "last_name", "domain"],
+                name="unique_ox_display_name",
+            ),
```

Comment:
Actually, I'm not totally convinced we should add this on our side, because it is really related to dimail/ox 
If you really feel it is mandatory to have this, then add a comment to explain, because I'm not sure we will remember this is only because of OX just by the name in 6 months ^^